c/zss.c

```diff
@@ -698,37 +698,94 @@ static void installWebPluginDefintionsService(WebPluginListElt *webPlugins, Http
   zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_DEBUG2, "end %s\n", __FUNCTION__);
 }
 
-static int checkLoggingVerbosity(const char *serverConfigFile, char *pluginIdentifier, ShortLivedHeap *slh) {
+#define ENV_LOGLEVELS_PREFIX "ZWED_logLevels_"
+
+static int getLogLevelsFromEnv(JsonObject *envConfig, char *identifier) {
+  int i = 0, j = 0;
+  int logLevel = ZOWE_LOG_SEVERE;
+  char envKey[2048] = {0};
+  j = snprintf(envKey, strlen(ENV_LOGLEVELS_PREFIX)+1, "%s", ENV_LOGLEVELS_PREFIX);
+  for (i = 0; identifier[i]; i++)
+  {
+    if (identifier[i] == '.') {
+      envKey[j++] = '_'; envKey[j++] = '_'; envKey[j++] = '_';
+    } else {
+      envKey[j++] = identifier[i];
+    }
+  }  
+  envKey[j] = '\0';
+
+  logLevel = jsonObjectGetNumber(envConfig, envKey);
+  return logLevel;
+}
+
+static int checkLoggingVerbosity(JsonObject *serverConfig, JsonObject *envConfig, char *pluginIdentifier) {
   char errorBuffer[1024] = {0};
-  
-  Json *json = jsonParseFile(slh, serverConfigFile, errorBuffer, sizeof(errorBuffer));
-  if (json) {
-    if (jsonIsObject(json)) {
-      JsonObject *jsonObject = jsonAsObject(json);
-      JsonObject *logLevels = jsonObjectGetObject(jsonObject, "logLevels");
-      if (logLevels != NULL) {
-        JsonProperty *property = jsonObjectGetFirstProperty(logLevels);
-        while (property != NULL) {
-          if (!strcmp(jsonPropertyGetKey(property), pluginIdentifier)) {
-            int logLevel = jsonObjectGetNumber(logLevels, pluginIdentifier);
-            if (logLevel <= ZOWE_LOG_DEBUG3 || logLevel >= ZOWE_LOG_ALWAYS) {
-              return logLevel;
-            }
-            else {
-              break;
-            }
-          }
-          property = jsonObjectGetNextProperty(property);
+  int logLevel = ZOWE_LOG_SEVERE;
+
+  logLevel = getLogLevelsFromEnv(envConfig, pluginIdentifier);
+  if (!logLevel) {
+    JsonObject *logLevels = jsonObjectGetObject(serverConfig, "logLevels");
+    if (logLevels != NULL) {
+      JsonProperty *property = jsonObjectGetFirstProperty(logLevels);
+      while (property != NULL) {
+        if (!strcmp(jsonPropertyGetKey(property), pluginIdentifier)) {
+          logLevel = jsonObjectGetNumber(logLevels, pluginIdentifier);
+          break;
         }
+        property = jsonObjectGetNextProperty(property);
       }
     }
   }
-  
+
+  if (logLevel <= ZOWE_LOG_DEBUG3 || logLevel >= ZOWE_LOG_ALWAYS) {
+    return logLevel;
+  }  
   return ZOWE_LOG_INFO;
 }
 
+static char *formatDataServiceIdentifier(const char *unformattedIdentifier, size_t identifierLength, char idNameSeparator)
+{
+  char *dataServiceIdentifier = safeMalloc31(identifierLength, "dataServiceIdentifier");
```

Comment:
`safeMalloc`

Reply:
Check if `dataServiceIdentifier` is not NULL

Reply:
Resolved